swift/ql/src/queries/Security/CWE-327/BrokenCryptoAlgorithm.ql

```diff
@@ -0,0 +1,30 @@
+/**
+ * @name Use of a broken or weak cryptographic algorithm
+ * @description Using broken or weak cryptographic algorithms can compromise security.
+ * @kind problem
+ * @problem.severity warning
+ * @security-severity 7.5
+ * @precision high
+ * @id swift/weak-cryptographic-algorithm
+ * @tags security
+ *       external/cwe/cwe-327
+ */
+
+import swift
+import codeql.swift.security.Cryptography // TODO: or use a megamodule like Concepts, containing a Cryptography module?
+
+from
+  Cryptography::CryptographicOperation operation, Cryptography::CryptographicAlgorithm algorithm,
+  string msgPrefix
+where
+  algorithm = operation.getAlgorithm() and
+  // `Cryptography::HashingAlgorithm` and `Cryptography::PasswordHashingAlgorithm` are
+  // handled by `py/weak-sensitive-data-hashing`
```

Comment:
```suggestion
  // Hashing algorithms are handled by `swift/weak-sensitive-data-hashing`
```
since notably `swift/weak-sensitive-data-hashing` does *not* use the Cryptography module (as hashing is provided in fairly standard libraries that can be modelled).

Reply:
This has been addressed.